scripts/zones/Arrapago_Reef/Zone.lua

```diff
@@ -18,27 +18,24 @@ end
 
 function onZoneIn(player,prevZone)
     local cs = -1
+    if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
+        player:setPos(-456, -3, -405, 64)
+    end
 
-    if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
-        if prevZone == dsp.zone.THE_ASHU_TALIF then
-            if (player:getCurrentMission(TOAU) == THE_BLACK_COFFIN and player:getVar("AhtUrganStatus") == 2) then
-                cs = 9
-            elseif player:getVar("AgainstAllOdds") == 3 then
-                player:setPos(-456, -3, -405, 64)
-                cs = 238
-            else
-                player:setPos(-456, -3, -405, 64)
-            end
-        elseif (prevZone == dsp.zone.CAEDARVA_MIRE and player:getCurrentMission(TOAU) == PREVALENCE_OF_PIRATES and player:getVar("AhtUrganStatus") == 0) then
-        cs = 13
-        else
-            player:setPos(-180.028,-10.335,-559.987,182)
+    if prevZone == dsp.zone.THE_ASHU_TALIF then
+        if player:getVar("AhtUrganStatus") == 2 and player:getCurrentMission(TOAU) == THE_BLACK_COFFIN then
```

Comment:
Not sure why you changed the order of conditions, it would make sense to check for the appropriate mission before it bothers checking `AhtUrganStatus`